### ISSUE TITLE
Improved ordering options

### ISSUE THREAD

**Issue by mjtamlyn** [orf/django-choice-object#3]
Hi,

I have a use case where I want to explicitly order the options but want to maintain "slugs" for the keys. I can do something horrible where we order by the display names but they're actually objects with `__str__` and `__cmp__` but that's horrible. It would be awesome to be able to define something like this:

```
class MyChoices(ChoiceObject):
    _order_key = 2
    APPLES = ('apples', 'Apples', 1)
    BANANAS = ('bananas', 'Bananas', 0)
```

i.e. if _order_key is set go with that instead of _order_by. Or just being able to set _order_by in this manner instead. We would need to keep the "original" data to sort on as well as the current dict in `_data`.

I'm happy to provide a PR for this if you think it's a nice idea.

**Comment by orf** [comment]
Heya,
I love the idea, I'll be happy to merge a PR that implements this. Perhaps we could use an ordered dict and [get rid of this bit entirely?](https://github.com/orf/django-choice-object/blob/master/django_choice_object/choice.py#L32). The ordering isn't dynamic and is known within the `__init__`, I'm not sure why I didn't just use an ordered dict to begin with. That would remove needing to store the original data along with `_data`.

Not sure how this would work with inheritance though.

**Comment by mjtamlyn** [comment]
It would definitely make more sense to use an `OrderedDict` for `_data`. I think I might add a `_get_sort_key(value)` method which you could override if you want to do complex things (like having some other objects than strings as the values), which I could easily override here to make the ordering based on the third part of the tuple. I think storing `cls._raw` could still be useful as an unsorted dict as `_data` would strip the sort order in this case so if I wanted to access that for some reason later on it would be lost completely (as the original attributes of the class don't exist).

**Comment by orf** [comment]
Fixed in #5
